Assets/Shopify/UIToolkit/ShopControllers/MultiProductShopController.cs

```diff
@@ -1,10 +1,40 @@
 ﻿namespace Shopify.UIToolkit {
     using System.Collections;
     using System.Collections.Generic;
+    using Shopify.Unity;
     using UnityEngine;
 
     public class MultiProductShopController : ShopControllerBase {
-        public override void OnHide() {}
-        public override void OnShow() {}
+        public new IMultiProductShop Shop { 
+            get { 
+                return base.Shop as IMultiProductShop; 
+            }
+
+            set {
+                base.Shop = value;
+            } 
+        }
+
+        public override void OnHide() {
+            throw new System.NotImplementedException();
+        }
+
+        public override void OnShow() {
+            Client.products(OnProductsLoaded);
+        }
```

Comment:
worth noting that we are going to remove the hide/show func from the shop controller and leave it as a view concern... we will fix this forward thought